data/src/main/java/io/gnosis/data/repositories/TokenRepository.kt

```diff
@@ -23,7 +23,7 @@ class TokenRepository(
             else
                 Balance(
                     TokenInfo(
```

Comment:
TokenType parsing should be done by adapter. Also this mapping step is not needed at all, as everything can be parsed directly from client gw response

Reply:
It shouldn't be necessary. But we would need a custom adapter to implement this logic. I would rather have the client gateway send a response that doesn't need this logic. 

If you remove this code, tests will fail.